### DISCUSSION TITLE
How to solve model rotation around a random point issue ?

### DISCUSSION THREAD

**Discussion by udayk07** [sceneview/sceneform-android#279] [Q&A]
We are trying to implement marker based detection in AR. While loading the model on marker detected, users can not rotate ( some specific models ) around its center instead it rotates around some random point ( it may be transform issue ). Is there any workaround available in sceneform to resolve this issue ?

Sample model:

[test_rotation_issue.zip](https://github.com/SceneView/sceneform-android/files/7837869/test_rotation_issue.zip)

Screen recording:
https://user-images.githubusercontent.com/78372917/148204264-82eea9ba-eabf-433e-a3f0-a70f8f99c6f9.mp4

Thanks in advance.

**Comment by RGregat** [comment]
A first quick look.
The origin of the model is not correct

![image](https://user-images.githubusercontent.com/3974162/148747589-58994910-c82b-4d68-915c-6be2837fec75.png)

**Comment by RGregat** [comment]
can you checkout this model?
[test_rotation_issue_changed_origin.glb.zip](https://github.com/SceneView/sceneform-android/files/7838083/test_rotation_issue_changed_origin.glb.zip)

**Reply by udayk07** [reply]
@RGregat Thanks for the quick response. The given orgin changed model works fine. 

Is there any workaround to handle these type of models in sceneform ( as those models are dynamically uploaded from users end ) ?

**Reply by grassydragon** [reply]
Yes, you can use the Filament bounding box to scale and center the model: https://github.com/google/filament/blob/main/android/filament-android/src/main/java/com/google/android/filament/Box.java

An example of accessing the Filament bounding box: https://github.com/SceneView/sceneform-android/discussions/243#discussioncomment-1832858
